### ISSUE TITLE
Themable command line cursor

### ISSUE THREAD

**Issue by creightor** [helix-editor/helix#11396]
In all of the light themes the cursor in the command line at the very bottom is invisible because both the background and cursor and white.
![image](https://github.com/user-attachments/assets/3670ba00-6cf1-4dc3-b966-baf60b26c1fb)
![image](https://github.com/user-attachments/assets/5ab5dd05-f0f1-4b50-b92d-eb7aa5b803f5)
This makes it hard to edit commands and AFAIK there is no way to configure the cursor color in that section, it would be nice if it was.

**Comment by CptPotato** [comment]
I think helix uses the terminal's cursor here and doesn't have control over the color.

**Comment by creightor** [comment]
> I think helix uses the terminal's cursor here and doesn't have control over the color.

The cursor is themeable in buffers though
![image](https://github.com/user-attachments/assets/a8801aa5-c018-4bdc-969c-e690db9ca6db)

**Comment by CptPotato** [comment]
Yes, the block cursor in buffers is not using the native terminal cursor, instead it simply changes the cells background color.

I think it should be possible to have the command line work the same, but it needs to be refactored to use the same cell coloring as buffers that use the block cursor.
